hm_map.js

```diff
@@ -0,0 +1,28 @@
+//Create a map
+function createMap(arr) {
+  let map = new Map(Object.entries(arr));
+  return map;
+}
+createMap([
+  { id: 1, value: 1, date: "2022-02-15" },
+  { id: 2, value: 1242, date: "2023-02-15" },
+  { id: 3, value: 3342, date: "2021-02-15" },
+]);
+
+//Create a cache
+//cacheCalcDateDiff(startDate, endDate);
+
+const cache = new Map();
+
+const cacheCalcDateDiff = (startDate, endDate) => {
+  const start = new Date(startDate); // string to obj date
+  const end = new Date(endDate); // string to obj date
+  const key = `${startDate} ${endDate}`; //generate unic key
+  if (!cache.has(key)) {
+    //if there is not key in cache
+    const diff = end.getTime() - start.getTime(); // calculate diff between end and start for get value
```

Comment:
ideally is to use function from date diff task not just 
`const diff = end.getTime() - start.getTime();`